core/common/modules/dev-tools/deprecation.php

```diff
@@ -0,0 +1,169 @@
+<?php
+namespace Elementor\Core\Common\Modules\DevTools;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class Deprecation {
+	const SOFT_VERSIONS_COUNT = 4;
+	const HARD_VERSIONS_COUNT = 8;
+
+	/**
+	 * @var Deprecation
+	 */
+	private $deprecation;
+
+	public function __construct() {
+		$this->deprecation = new Deprecation();
+	}
+
+	/**
+	 * Get total of major.
+	 *
+	 * @param object $parsed_version
+	 *
+	 * @return int
+	 */
+	private function get_total_major( $parsed_version ) {
+		$major2 = $parsed_version->major2;
+		$major2 = $major2 > 9 ? 9 : $major2;
+		$minor = 0;
+
+		$total = intval( "{$parsed_version->major1}{$major2}{$minor}" );
+
+		if ( $total > 99 ) {
+			$temp = $total / 10;
+			$total = $temp ? $temp : intval( substr( $total, 0, 2 ) );
+		} else {
+			$total = intval( $total / 10 );
+		}
+
+		return $total;
+	}
+
+	/**
+	 * Get's next version.
+	 *
+	 * @param string $version
+	 * @param int $count
+	 *
+	 * @return string|false
+	 */
+	public function get_next_version( $version, $count = 1 ) {
+		$version = $this->parse_version( $version );
+
+		if ( ! $version ) {
+			return false;
+		}
+
+		$version->total = $this->get_total_major( $version );
+
+		$version->total += $count;
+
+		if ( $version->total > 9 ) {
+			$version->major1 = intval( $version->total / 10 );
+			$version->major2 = intval( $version->total % 10 );
+		} else {
+			$version->major1 = 0;
+			$version->major2 = $version->total;
+		}
+
+		$version->minor = 0;
+
+		return $this->implode_version( $version );
+	}
+
+	public function get_soft_deprecated_version( $version ) {
+		return $this->get_next_version( $version, Module::SOFT_VERSIONS_COUNT );
+	}
+
+	public function get_hard_deprecated_version( $version ) {
+		return $this->get_next_version( $version, Module::HARD_VERSIONS_COUNT );
+	}
+
+	/**
+	 * Implode parsed version to string version.
+	 *
+	 * @param object $parsed_version
+	 *
+	 * @return string
+	 */
+	public function implode_version( $parsed_version ) {
+		return "{$parsed_version->major1}.{$parsed_version->major2}.{$parsed_version->minor}";
+	}
+
+	/**
+	 * Parse version('0.0.0') to informative object.
+	 *
+	 * @param string $version
+	 *
+	 * @return object|false
+	 */
+	public function parse_version( $version ) {
+		$version_explode = explode( '.', $version );
+
+		if ( count( $version_explode ) != 3 ) {
+			trigger_error('Invalid Semantic Version string provided' );
+			return false;
+		}
+
+		list( $major1, $major2, $minor ) = $version_explode;
+
+		return ( object ) [
+			'major1' => intval( $major1 ),
+			'major2' => intval( $major2 ),
+			'minor' => intval( $minor ),
+		];
+	}
+
+	/**
+	 * Compare two versions, result is equal to diff of major versions.
+	 *
+	 * @param {string} $version1
+	 * @param {string} $version2
+	 *
+	 * @return int|false
+	 */
+	public function compare_version( $version1, $version2 ) {
```

Comment:
This method will return the wrong answer for any version with a "major2" higher than 9

Reply:
Handled.